py/torch_tensorrt/dynamo/_refit.py

```diff
@@ -451,17 +451,24 @@ def refit_module_weights(
                     weight_name_map=None,
                 )
 
+        # clear EXCLUDE_WEIGHTS flag
+        serialization_config = engine.create_serialization_config()
+        serialization_config.clear_flag(trt.SerializationFlag.EXCLUDE_WEIGHTS)
+        serialized_engine = engine.serialize_with_config(serialization_config)
+        engine = runtime.deserialize_cuda_engine(serialized_engine)
+
+        if isinstance(compiled_submodule, PythonTorchTensorRTModule):
+            compiled_submodule.engine = engine
+
         if isinstance(compiled_submodule, TorchTensorRTModule):
-            serialized_engine = bytes(engine.serialize())
             new_engine_info = list(engine_info)
-            new_engine_info[ENGINE_IDX] = serialized_engine
+            new_engine_info[ENGINE_IDX] = bytes(serialized_engine)
```

Comment:
Seems like we only need to deserialize in a PythonTorchTensorRTModule and we should probably use setup_engine instead. The standard interface should be like provide the serialized engine, then setup engine will set the module up properly

Reply:
> Seems like we only need to deserialize in a PythonTorchTensorRTModule

@narendasan Did you mean we only deserialize if `compiled_submodule` is a `PythonTorchTensorRTModule`? why not deserialize for `TorchTensorRTModule`?

Reply:
modified, please take a look again. thanks